terraform/autoneg/main.tf

```diff
@@ -0,0 +1,335 @@
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "clusterrolebinding_autoneg_proxy_rolebinding" {
+  metadata {
+    name = "autoneg-proxy-rolebinding"
+    labels = {
+      app = "autoneg"
+    }
+    annotations = {}
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.clusterrole_autoneg_proxy_role.metadata[0].name
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = kubernetes_service_account.service_account.metadata[0].name
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+  }
+}
+
+resource "kubernetes_service" "service_autoneg_controller_manager_metrics_service" {
+  metadata {
+    annotations = {
+      "prometheus.io/port"   = "8443"
+      "prometheus.io/scheme" = "https"
+      "prometheus.io/scrape" = "true"
+      "cloud.google.com"     = "'{\"ingress\":true}'"
+    }
+    labels = {
+      "app"           = "autoneg"
+      "control-plane" = "controller-manager"
+    }
+    name      = "autoneg-controller-manager-metrics-service"
+    namespace = "autoneg-system"
+  }
+  spec {
+    port {
+      name        = "https"
+      port        = 8443
+      target_port = "https"
+      protocol    = "TCP"
+    }
+    selector = {
+      app           = "autoneg"
+      control-plane = "controller-manager"
+    }
+  }
+  depends_on = [kubernetes_namespace.namespace_autoneg_system]
+}
+
+resource "kubernetes_deployment" "deployment_autoneg_controller_manager" {
+  metadata {
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+    name      = "autoneg-controller-manager"
+    labels = {
+      "app"           = "autoneg"
+      "control-plane" = "controller-manager"
+    }
+    annotations = {}
+  }
+
+  spec {
+    replicas = 1
+    selector {
+      match_labels = {
+        app           = "autoneg"
+        control-plane = "controller-manager"
+      }
+    }
+
+
+    template {
+      metadata {
+        labels = {
+          app           = "autoneg"
+          control-plane = "controller-manager"
+        }
+        annotations = {}
+      }
+
+      spec {
+        service_account_name             = kubernetes_service_account.service_account.metadata[0].name
+        termination_grace_period_seconds = 10
+
+        container {
+          name = "manager"
+
+          image             = var.controller_image
+          image_pull_policy = var.image_pull_policy
+
+          args    = ["--metrics-addr=127.0.0.1:8080", "--enable-leader-election"]
+          command = ["/manager"]
+
+          resources {
+            limits = {
+              cpu    = "100m"
+              memory = "30Mi"
+            }
+            requests = {
+              cpu    = "100m"
+              memory = "20Mi"
+            }
+          }
+        }
+
```

Comment:
Add `security_context` for both of these containers (`run_as_user`/`run_as_non_root`/`privileged`/etc) to allow security evaluations to pass (PodSecurityPolicy/OPA Gatekeeper/other admission hooks)

Reply:
See below for example `securityContext` settings: https://github.com/GoogleCloudPlatform/gke-autoneg-controller/blob/b791fecca246635f1aaa8ba790e12b0bea249421/deploy/autoneg.yaml#L199-L231

Reply:
Adding security context.